File: BC - w4/Car/car.cpp
#include <iostream>
#include "Point.h"
#include "Car.h"


        Car::Car(double capacity, double consumption, const Point& location, const std::string& model) :
        fuelCapacity(capacity),
        fuelConsumption(consumption),
        location(location),
        model(model),
        fuelAmount(0)
        {}

        Car::~Car(){}

        double Car::getFuelAmount() const{
            return this->fuelAmount;

        }
        double Car::getFuelCapacity() const{
            return this->fuelCapacity;

        }
        double Car::getFuelConsumption() const{
            return this->fuelConsumption;

        }
        const Point& Car::getLocation() const{
            return this->location;

        }
        const std::string& Car::getModel() const{
            return this->model;

        }

        void Car::drive(const Point& destination) {
        double canGet;
        double distance;

        distance = location.distance(destination);
        canGet = distance * fuelConsumption;
        if ( canGet > fuelAmount ) {
            throw OutOfFuel();
        }
        fuelAmount-=canGet;
        location = destination;
        }


        void Car::drive(double x, double y){
            drive(Point(x,y));
        }

        void Car::refill(double fuel){
            double NewFuelAmount = fuelAmount;

            NewFuelAmount += fuel;
            if (NewFuelAmount > fuelCapacity) {
                throw ToMuchFuel();
            }
            fuelAmount = NewFuelAmount;
        }

        std::ostream& operator<<(std::ostream& out, const Car& car) {
            out <<"fuel Capacity " << car.getFuelCapacity() << std::endl;
            out <<"consumption " << car.getFuelConsumption() << std::endl;
            out <<"fuel Amount " << car.getFuelAmount() << std::endl;
            out <<"location " << car.getLocation();
            out <<"model " << car.getModel() << std::endl;
        }

File: BC - w4/java/Car/Car.java
package Car;


public class Car {
	
	private double fuelAmount;
	private double fuelCapacity;
	private double fuelConsumption;
	private Point location;
	private String model;
	
    public Car(double capacity, double consumption,  Point location, String model){
    	this.fuelCapacity = capacity;
    	this.fuelConsumption = consumption;
    	this.location = location;
    	this.model = model;
    	this.fuelAmount = 0;
    }


    public double getFuelAmount() {
        return this.fuelAmount;

    }
    public double getFuelCapacity() {
        return this.fuelCapacity;
        }
    public double getFuelConsumption() {
        return this.fuelConsumption;
    }
    
    public Point getLocation() {
        return this.location;
        }
        
    public String getModel() {
        return this.model;

    }

    public void drive( Point destination) throws OutOfFuel{
    double canGet;
    double distance;

    distance = location.distance(destination);
    canGet = distance * fuelConsumption;
    if ( canGet > fuelAmount ) {
        throw new OutOfFuel();
    }
    fuelAmount-=canGet;
    location = destination;
    }


    public void drive(double x, double y)throws OutOfFuel {
        drive(new Point(x,y));
    }


	public void refill(double fuel) throws ToMuchFuel{
        double NewFuelAmount = fuelAmount;

        NewFuelAmount += fuel;
        if (NewFuelAmount > fuelCapacity) {
            throw new ToMuchFuel();
        }
        fuelAmount = NewFuelAmount;
    }
    
    
    public void print(){
		System.out.println("fuel Capacity "+ this.fuelCapacity+"\n"
							+"consumption " + this.fuelConsumption+"\n"
							+"fuel Amount " + this.fuelAmount+ "\n"
							+"location " + this.location + "\n"
							+"model " + this.model);
	}


}
